x-pack/plugins/security/server/config.ts

```diff
@@ -120,6 +129,38 @@ const providersConfigSchema = schema.object(
         schema.object({ ...getCommonProviderSchemaProperties(), realm: schema.string() })
       )
     ),
+    anonymous: getUniqueProviderSchema(
+      'anonymous',
+      {
+        description: schema.string({
+          defaultValue: i18n.translate('xpack.security.loginAsGuestLabel', {
+            defaultMessage: 'Continue as Guest',
+          }),
+        }),
+        hint: schema.string({
+          defaultValue: i18n.translate('xpack.security.loginAsGuestHintLabel', {
+            defaultMessage: 'For anonymous users',
+          }),
+        }),
+        icon: schema.string({ defaultValue: 'globe' }),
+        session: schema.object({
+          idleTimeout: schema.nullable(schema.duration()),
```

Comment:
__note:__ having `nullable` here means that `idleTimeout` for anonymous provider can be non-null only if it's specifically reconfigured to another non-null value.